cpu/lm4f120/periph/timer.c

```diff
@@ -35,132 +37,411 @@
+      low_bits = (unsigned long long)ROM_TimerValueGet(timer_base, timer_side);
+      high_bits_dup = ((unsigned long long)PRIV_TimerPrescaleSnapshotGet(timer_base, timer_side)) << 32;
+    } while (high_bits != high_bits_dup);
+
+    unsigned long long total = high_bits + low_bits;
+    DEBUG("Combined %lx:%lx\n", (unsigned long) (total>>32), (unsigned long) (total & 0xFFFFFFFF));
+
+    unsigned int scaled_value = _llvalue_to_scaled_value(total, config[dev].diviser);
+
+    return scaled_value;
 }
 
 void timer_start(tim_t dev)
 {
-    if (dev == TIMER_0) {
-        ROM_TimerEnable(WTIMER0_BASE, TIMER_A);
+    unsigned int timer_base;
+    unsigned int timer_side = TIMER_A;
+
+    if (dev >= TIMER_NUMOF){
+        return ;
+    }
+
+    switch(dev){
+#if TIMER_0_EN
+    case TIMER_0:
+        timer_base = WTIMER0_BASE;
+        break;
+#endif
+#if TIMER_1_EN
+    case TIMER_1:
+        timer_base = WTIMER1_BASE;
+        break;
+#endif
+    default:
+        return; // unreachable
     }
+
+    ROM_TimerEnable(timer_base, timer_side);
 }
 
 void timer_stop(tim_t dev)
 {
-    if (dev == TIMER_0) {
-        ROM_TimerDisable(WTIMER0_BASE, TIMER_A);
+    unsigned int timer_base;
+    unsigned int timer_side = TIMER_A;
+
+    if (dev >= TIMER_NUMOF){
+        return;
     }
+
+    switch(dev){
+#if TIMER_0_EN
+    case TIMER_0:
+        timer_base = WTIMER0_BASE;
+        break;
+#endif
+#if TIMER_1_EN
+    case TIMER_1:
+        timer_base = WTIMER1_BASE;
+        break;
+#endif
+    default:
+        return; // unreachable
+    }
+
+    ROM_TimerDisable(timer_base, timer_side);
 }
 
 void timer_irq_enable(tim_t dev)
 {
-    if (dev == TIMER_0) {
-        ROM_IntEnable(INT_WTIMER0A);
-        ROM_TimerIntEnable(WTIMER0_BASE, TIMER_TIMA_TIMEOUT);
+    if (dev >= TIMER_NUMOF){
+        return;
     }
+
+    unsigned int timer_intbase;
+
+    switch(dev){
+#if TIMER_0_EN
+    case TIMER_0:
+        timer_intbase = INT_WTIMER0A;
+        break;
+#endif
+#if TIMER_1_EN
+    case TIMER_1:
+        timer_intbase = INT_WTIMER1A;
+        break;
+#endif
+    default:
+        return; // unreachable
+    }
+
+    ROM_IntPrioritySet(timer_intbase, 32);
+    ROM_IntEnable(timer_intbase);
 }
 
 void timer_irq_disable(tim_t dev)
 {
-    if (dev == TIMER_0) {
-        ROM_IntDisable(INT_WTIMER0A);
+    if (dev >= TIMER_NUMOF){
+        return;
+    }
+    unsigned int timer_base;
+    unsigned int timer_intbit = TIMER_TIMA_TIMEOUT;
+    unsigned int timer_intbase;
+
+    switch(dev){
+#if TIMER_0_EN
+    case TIMER_0:
+        timer_base = WTIMER0_BASE;
+        timer_intbase = INT_WTIMER0A;
+        break;
+#endif
+#if TIMER_1_EN
+    case TIMER_1:
+        timer_base = WTIMER1_BASE;
+        timer_intbase = INT_WTIMER1A;
+        break;
+#endif
+    default:
+        return; // unreachable
     }
+
+    ROM_IntEnable(timer_intbase);
+    ROM_TimerIntDisable(timer_base, timer_intbit);
 }
 
 void timer_reset(tim_t dev)
```

Comment:
Yes, that's already been taken care of in my local rebase (see previous comment).